crates/core/src/config/mod.rs

```diff
@@ -181,6 +181,49 @@ impl Config {
     }
 }
 
+pub mod compression_format {
```

Comment:
This part feels a little "too much," so if you have any improvements here @tarkah, please let me know.
Essentially I have extended `CompressionFormat::Zstd` to take an `i32` now, so I need a migration from `Zstd` -> `Zstd(i32)` in order not to break config.

What I have now works, but it just feels like a lot of code for such a small change—damn migrations.